### ISSUE TITLE
Shorten settings-titles

### ISSUE THREAD

**Issue by Glitchy-Tozier** [florisboard/florisboard#998]
I think it's best if users didn't have to turn their phone sideways, just to know what an option does.

I suggest either
- giving titles line-breaks (the easy solution) or
- working through them in the languages available tp you and thinking on how to name them more efficiently. (probably the more difficult but better solution.

Motivation:
![Screenshot_20210605-065254_FlorisBoard Beta.jpg](https://user-images.githubusercontent.com/59611881/120900136-7c8a4400-c633-11eb-8444-5d1aee048890.jpg)

**Comment by patrickgold** [comment]
> I think it's best if users didn't have to turn their phone sideways, just to know what an option does.

Ah the beauty of the German language again :) I am impressed how these short & shiny English titles look soo huge and long in German.

> giving titles line-breaks (the easy solution) or

That's against the Material Design guidelines, a title should never have a line break in it.

> working through them in the languages available tp you and thinking on how to name them more efficiently. (probably the more difficult but better solution.

Yes, that's definitely the better option. I can at least do it for German when I find the time for it. Though I fear we may have some cases where there simply is no shorter option available. That will be fun to find something different then :)

**Comment by patrickgold** [comment]
> Ew really? I'll have to do a lot of thinking for my current (unrelated) project then.

I just looked it up and they don't specifically mention that a two-line title is forbidden, but they define 3 types of how a list can look like: https://material.io/components/lists#types

These three types are:
- 1 title line / no summary
- 1 title line / 1 summary line
- 1 title line / 2 summary lines

Based on this I am concluding that two-line titles should not be used when following Material guidelines. (I could be wrong though but that's how I understood it).

> my current (unrelated) project then

As you mentioned several times that you are working on your own project, I am a bit curious what it is :) (if you don't mind sharing it of course)

>  Hit me up when you start working on it, i might be able to give feedback :)

Will do when  I start working on this!

**Comment by Glitchy-Tozier** [comment]
> I just looked it up and they don't specifically mention that a two-line title is forbidden, but they define 3 types of how a list can look like: https://material.io/components/lists#types

Thank you very much for sharing those things with me, I didn't even know there were preferred list-designs!

> As you mentioned several times that you are working on your own project, I am a bit curious what it is :) (if you don't mind sharing it of course)

Ah, right, I asked you about Android Studio!
Well, since about ~2 weeks I am working on a fun little app for myself. It's basically an to-do list that 
1. only has one to-do a day
2. incrementally moves you closer to some end-goal.

For example, if you wanted to ... go jogging every morning, for a week the to-do would be to just leave the house and return.
Then the to-do would be to walk around your house (each day for a week)
• Walk around 3-5 buildings
• Jog around 3-5 buildings
• Jog for a longer time. Yay, now you're jogging every day.

Does my explanation make sense?

My main motivation is to write it for myself, but i plan on sending it to PlayStore/Fdroid when the app is in an acceptable state. The main code already is available on Github, just not everything yet as i still haven't figured out how Github works.  xD

Also, Flutter's fun. It feels inconsistent compared to how nice CSS is, but it has taught me a lot, especially about Object Oriented Programming (which i almost had zero knowledge in).

**Comment by patrickgold** [comment]
Thanks for sharing, this sounds like a nice little project :)

> Also, Flutter's fun

Flutter is fun and it is cross-platform, especially at the learning phase! I once built some local projects in it as well and it was the entry to Android at the time. Then I learned Kotlin (via FlorisBoard), and it is even more fun :) Kotlin is a very thought through language and wraps ugly Java conventions in easy-to-use functions and statements. (like the `when` statement, classes look cleaner, extension functions, etc.).

> My main motivation is to write it for myself, but i plan on sending it to PlayStore/Fdroid when the app is in an acceptable state

Tell me when you do and I will try it out :)

**Comment by Glitchy-Tozier** [comment]
> Then I learned Kotlin (via FlorisBoard), and it is even more fun :) Kotlin is a very thought through language and wraps ugly Java conventions in easy-to-use functions and statements.

No idea what java is like, but if I'll ever devellop a native App, I'll definitely go for Kotlin!

> Tell me when you do and I will try it out :)

Will do, looking forward to your thoughts  :)

**Comment by patrickgold** [comment]
In the whole development for 0.3.14 I tried to shorten the settings titles and summaries, above commit is an additional fix for it. I see this issue as semi-fixed now, because I shortened the titles wherever possible and in general there are minor occurrences where there are still ellipses.

Note that this only applies to the original source strings for English, I don't know the state of other languages. Thanks again for reporting this issue!
